lib/floki/text_extractor.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Floki.TextExtractor do
+  @moduledoc false
+
+  @allowed_input_types [
+    nil,
+    "color",
+    "date",
+    "datetime-local",
+    "email",
+    "month",
+    "number",
+    "search",
+    "tel",
+    "text",
+    "time",
+    "url",
+    "week"
+  ]
+
+  def extract_input_value(attrs) do
+    type = Enum.find(attrs, &match?({"type", _}, &1))
+
+    case type do
+      {"type", t} ->
+        if t in @allowed_input_types do
+          extract_value(attrs)
+        else
+          ""
+        end
+
+      nil ->
```

Comment:
If we couldn't find the input type, shouldn't we just return `nil`? Or `nil` type implies that this is a `text` input? :thinking:

Reply:
I've simplified this part to default an input's type to `text` if it doesn't have the attribute, I believe the code is clearer now